toolkits/github/arcade_github/tools/activity.py

```diff
@@ -38,3 +38,56 @@ async def set_starred(
 
     action = "starred" if starred else "unstarred"
     return f"Successfully {action} the repository {owner}/{name}"
+
+
+# Implements https://docs.github.com/en/rest/activity/starring?apiVersion=2022-11-28#list-stargazers
+# Example `arcade chat` usage: "list the stargazers for the ArcadeAI/arcade-ai repo"
+@tool(requires_auth=GitHub())
+async def list_stargazers(
+    context: ToolContext,
+    owner: Annotated[str, "The owner of the repository"],
+    repo: Annotated[str, "The name of the repository"],
+    limit: Annotated[
+        Optional[int],
+        "The maximum number of stargazers to return. If not provided, all stargazers will be returned.",
+    ] = None,
+) -> Annotated[dict, "A dictionary containing the stargazers for the specified repository"]:
+    """List the stargazers for a GitHub repository."""
+    url = get_url("repo_stargazers", owner=owner, repo=repo)
+    headers = get_github_json_headers(context.authorization.token)
+
+    per_page = 30
+    page = 1
+    stargazers = []
+
+    if limit is None:
+        limit = 2**64 - 1
+
+    async with httpx.AsyncClient() as client:
+        while len(stargazers) < limit:
+            response = await client.get(
+                url, headers=headers, params={"per_page": per_page, "page": page}
+            )
+            handle_github_response(response, url)
+
+            data = response.json()
+            if not data:
+                break
+
+            stargazers.extend([
+                {
+                    "login": stargazer.get("login"),
+                    "id": stargazer.get("id"),
+                    "node_id": stargazer.get("node_id"),
+                    "html_url": stargazer.get("html_url"),
+                }
+                for stargazer in data
+            ])
+
+            if len(data) < per_page:
+                break
```

Comment:
If I'm reading this right, this will sometimes result in `len(data) > limit` because it adds a full page of results every time, and _then_ checks that it's over the limit.

Reply:
yeah, if
```
if len(data) < per_page:
    break
```
is satisfied, then this means that there are less stargazers than the requested limit.

Otherwise, if there are equal or more stargazers than the requested limit, then the `while` condition will no longer be satisfied.

Reply:
I gotcha, I missed this crucial bit below
```
stargazers = stargazers[:limit]
```

😅